src/mbgl/mtl/drawable.cpp

```diff
@@ -364,103 +344,85 @@ void Drawable::setVertexAttrNameId(const StringIdentity value) {
     impl->idVertexAttrName = value;
 }
 
-void Drawable::bindAttributes(const RenderPass& renderPass) const {
+void Drawable::bindAttributes(RenderPass& renderPass) const noexcept {
     const auto& encoder = renderPass.getMetalEncoder();
 
     NS::UInteger attributeIndex = 0;
     for (const auto& binding : impl->attributeBindings) {
-        if (const auto buffer = getMetalBuffer(binding ? binding->vertexBufferResource : nullptr)) {
+        const auto* buffer = static_cast<const mtl::VertexBufferResource*>(binding ? binding->vertexBufferResource
+                                                                                   : nullptr);
+        if (buffer && buffer->get()) {
             assert(binding->vertexStride * impl->vertexCount <= getBufferSize(binding->vertexBufferResource));
-            encoder->setVertexBuffer(buffer, /*offset=*/0, attributeIndex);
+            renderPass.bindVertex(buffer->get(), /*offset=*/0, attributeIndex);
         } else if (impl->noBindingBuffer) {
-            encoder->setVertexBuffer(impl->noBindingBuffer, /*offset=*/0, attributeIndex);
+            renderPass.bindVertex(impl->noBindingBuffer->get(), /*offset=*/0, attributeIndex);
         }
         attributeIndex += 1;
     }
 }
 
-void Drawable::unbindAttributes(const RenderPass& renderPass) const {
-    const auto& encoder = renderPass.getMetalEncoder();
-
-    NS::UInteger attributeIndex = 0;
-    for (const auto& binding : impl->attributeBindings) {
-        encoder->setVertexBuffer(nullptr, /*offset=*/0, attributeIndex);
-        attributeIndex += 1;
-    }
-}
-
-void Drawable::bindUniformBuffers(const RenderPass& renderPass) const {
+void Drawable::bindUniformBuffers(RenderPass& renderPass) const noexcept {
     if (shader) {
         const auto& shaderMTL = static_cast<const ShaderProgram&>(*shader);
         const auto& encoder = renderPass.getMetalEncoder();
         for (const auto& element : shaderMTL.getUniformBlocks().getMap()) {
             const auto& uniformBuffer = getUniformBuffers().get(element.first);
             if (!uniformBuffer) {
-                using namespace std::string_literals;
-                const auto tileID = getTileID() ? util::toString(*getTileID()) : "<no tile>";
-                const auto tileLabel = util::toString(getID()) + "/" + getName() + "/" + tileID;
-                Log::Error(Event::General,
-                           "bindUniformBuffers: UBO "s + stringIndexer().get(element.first) + " not found on " +
-                               tileLabel + ". skipping.");
+                try { // discard logging exceptions to conform to `noexcept`
+                    using namespace std::string_literals;
+                    const auto tileID = getTileID() ? util::toString(*getTileID()) : "<no tile>";
+                    const auto tileLabel = util::toString(getID()) + "/" + getName() + "/" + tileID;
+                    Log::Error(Event::General,
```

Comment:
It would be a bug, but it might be in a custom layer implementation.  It would be nice to get useful feedback without a debug build, but maybe not worth it since we need to reduce code size.